frontend/src/libs/components/input/input.tsx

```diff
@@ -42,19 +46,29 @@ const Input = <T extends FieldValues>({
     hasError && styles.error,
   ];
 
+  const toggleShowPassword = useCallback((): void => {
+    setShowPassword(!showPassword);
+  }, [showPassword]);
+
   return (
     <label className={styles.inputComponentWrapper}>
       {hasLabel && <span className={styles.label}>{label}</span>}
       <span className={styles.inputWrapper}>
         <input
           {...field}
-          type={type}
+          type={showPassword ? 'text' : type}
           placeholder={placeholder}
           className={getValidClassNames(...inputStyles)}
           disabled={isDisabled}
         />
         {type === 'password' && (
-          <span className={styles.passwordEye}>&#128065;</span>
+          <input
```

Comment:
can we add it as button, not as input type="button"? 
also what about changing eye icon when turning on/off? now it's not working this way

Reply:
fixed